src/github.com/getlantern/flashlight/config/bootstrap.go

```diff
@@ -81,26 +81,39 @@ func readSettingsFromFile(yamlPath string) (*BootstrapSettings, error) {
 
 // MakeInitialConfig save baked-in config to the file specified by configPath
 func MakeInitialConfig(configPath string) error {
-	src, err := bootstrapPath(lanternYamlName)
+	dir, _, err := bootstrapPath(lanternYamlName)
+	if err != nil {
+		log.Errorf("Could not get bootstrap path %v", err)
+		return err
+	}
+
+	// We need to use tarfs here because the lantern.yaml needs to embedded
+	// in the binary for auto-updates to work. We also want the flexibility,
+	// however, to embed it in installers to change various settings.
+	fs, err := tarfs.New(Resources, dir)
 	if err != nil {
+		log.Errorf("Could not read resources? %v", err)
 		return err
 	}
-	bytes, err := ioutil.ReadFile(src)
+
+	bytes, err := fs.Get("lantern.yaml")
 	if err != nil {
+		log.Errorf("Could not read bootstrap file %v", err)
 		return err
 	}
 	err = ioutil.WriteFile(configPath, bytes, 0644)
```

Comment:
I guess this is the gist: we first check if we have a good config. Otherwise we extract one from the tarfs system and save it as a regular file. Next time we run Lantern, it will read that one.
What happens with autoupdates then?